examples/vae_text/vae_train.py

```diff
@@ -202,14 +200,36 @@ def _embedding_fn(token):
 
         return ret
 
+    def _embed_fn_rnn(self, tokens: torch.LongTensor) -> Tensor:
+        r"""Generates word embeddings
+        """
+        embedding = self.decoder_w_embedder(
+            tokens)
+        latent_z = self._latent_z
+        return torch.cat([embedding, latent_z], dim=-1)
+
+    def _embed_fn_transformer(self,
+                              tokens: torch.LongTensor,
+                              positions: torch.LongTensor) -> Tensor:
+        r"""Generates word embeddings combined with positional embeddings
+        """
+        output_p_embed = self.decoder_p_embedder(positions)
+        output_w_embed = self.decoder_w_embedder(tokens)
+        output_w_embed = output_w_embed * \
+            self._config.hidden_size ** 0.5
+        output_embed = output_w_embed + output_p_embed
+        return output_embed
+
     def decode(self,
                decoder_states: Union[Tensor, Tuple],
                helper: Optional[Helper],
+               latent_z: Optional[Tensor] = None,
                seq_lengths: Optional[Tensor] = None,
                max_decoding_length: int = None,
                text_ids: torch.LongTensor = None) \
```

Comment:
These type annotations will not pass type-check.

Reply:
I see, will correct `max_decoding_length ` and `text_ids `.